src/system-health/health_checker/service_checker.py

```diff
@@ -15,6 +15,19 @@
 EVENTS_PUBLISHER_SOURCE = "sonic-events-host"
 EVENTS_PUBLISHER_TAG = "process-not-running"
 
+def check_docker_image(image_name):
+    """
+    @summary: This function will check if docker image exists.
+    @return:  True if the image exists, otherwise False.
+    """
+    try:
+        DOCKER_CLIENT = docker.DockerClient(base_url='unix://var/run/docker.sock')
+        DOCKER_CLIENT.images.get(image_name)
```

Comment:
Is there an assumption that images.get function will drop an exception when image is not there? Should we also check what it returns other than depend the logic of images.get function? For example, if the images.get refined to return a null for a search miss, it would still return True by check_docker_image.

Reply:
This function will raise exception if the image does not exist:

https://docker-py.readthedocs.io/en/stable/images.html
get(name)[¶](https://docker-py.readthedocs.io/en/stable/images.html#docker.models.images.ImageCollection.get)

Raises:
docker.errors.ImageNotFound – If the image does not exist.

docker.errors.APIError – If the server returns an error.